formats/protobuf/commonMain/src/kotlinx/serialization/protobuf/internal/ProtobufDecoding.kt

```diff
@@ -331,6 +354,113 @@ private class MapEntryReader(
         else ProtoDesc(2, (parentTag.integerType))
 }
 
+private class OneOfPolymorphicReader(
+    proto: ProtoBuf,
+    decoder: ProtobufReader,
+    private val parentTag: ProtoDesc,
+    descriptor: SerialDescriptor
+) : ProtobufDecoder(proto, decoder, descriptor) {
+    private var serialNameDecoded = false
+    private var contentDecoded = false
+    override fun SerialDescriptor.getTag(index: Int): ProtoDesc = if (index == 0) {
+        POLYMORPHIC_NAME_TAG
+    } else {
+        extractParameters(index).overrideId(parentTag.protoId)
+    }
+
+    override fun beginStructure(descriptor: SerialDescriptor): CompositeDecoder {
+        return if (descriptor == this.descriptor) {
+            this
+        } else {
+            OneOfElementReader(proto, reader, descriptor)
+        }
+    }
+
+    override fun decodeElementIndex(descriptor: SerialDescriptor): Int {
+        if (!serialNameDecoded) {
+            serialNameDecoded = true
+            return 0
+        } else if (!contentDecoded) {
+            contentDecoded = true
+            return 1
+        } else {
+            return CompositeDecoder.DECODE_DONE
+        }
+    }
+
+    override fun decodeTaggedString(tag: ProtoDesc): String = if (tag == POLYMORPHIC_NAME_TAG) {
+        // todo return a serial name for polymorphic serializer
```

Comment:
In my previous commit, I would like to print both the class name and field name in the error message, but failed to get the field here, and just left some unclear comment.

Now it is just some unreachable code to avoid compilation issue, as explained in the new comment.